lib/util/string_conv/string_conv.c

```diff
@@ -0,0 +1,156 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <ctype.h>
+#include <string.h>
+#include <stdlib.h>
+#include <logging/log.h>
+
+#define LOG_MODULE_NAME string_conv
+LOG_MODULE_REGISTER(LOG_MODULE_NAME, CONFIG_STRING_CONV);
+
+int string_conv_str2long(char *str, int *val)
+{
+	int str_len = strlen(str);
+	int temp_val;
+	int idx = 0;
+
+	if (!strncmp(str, "-", 1)) {
+		idx = 1;
+	}
+
+	for (int i = idx; i < str_len; i++) {
+		if (isdigit((int)str[i])) {
```

Comment:
`strlen` in line 17 is not needed as the end of string may be found in loop 25, this way preventing looping twice the same range of `str`.
`strncmp` for single character seems overkill.